### PR TITLE
[SPARK-38678][TESTS][FOLLOWUP] Enable RocksDB tests in `AppStatusStoreSuite` and `StreamingQueryStatusListenerSuite` on Apple Silicon on MacOS

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to enable RocksDB tests in `AppStatusStoreSuite` and `StreamingQueryStatusListenerSuite` on Apple Silicon on MacOS, it's a followup of SPARK-38678.

### Why are the changes needed?
Enable more RocksDB related test on Apple Silicon on MacOS

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?

- Pass GA
- Manual test on Apple Silicon environment：

```
build/sbt "core/testOnly *AppStatusStoreSuite*" "sql/testOnly *StreamingQueryStatusListenerSuite*"
```

All tests passed.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/status/AppStatusStoreSuite.scala [hunk 1]
```diff
@@ -106,12 +102,22 @@ class AppStatusStoreSuite extends SparkFunSuite {
...
-    "in memory" -> createAppStore(disk = false, live = false),
-    "in memory live" -> createAppStore(disk = false, live = true)
-  ).foreach { case (hint, appStore) =>
+  private val cases = {
+    val baseCases = Seq(
+      "disk rocksdb" -> createAppStore(disk = true, HybridStoreDiskBackend.ROCKSDB, live = false),
+      "in memory" -> createAppStore(disk = false, live = false),
+      "in memory live" -> createAppStore(disk = false, live = true)
+    )
+    if(Utils.isMacOnAppleSilicon) {
```
Comment (dongjoon-hyun) [inline core/src/test/scala/org/apache/spark/status/AppStatusStoreSuite.scala:111]: nit. space.
```scala
- if(Utils.isMacOnAppleSilicon) {
+ if (Utils.isMacOnAppleSilicon) {
```

**Review by dongjoon-hyun — APPROVED** [review]
One one minor comment.

File: core/src/test/scala/org/apache/spark/status/AppStatusStoreSuite.scala (hunk 1 above)
Comment (LuciferYang) [reply core/src/test/scala/org/apache/spark/status/AppStatusStoreSuite.scala]: done

**Comment by LuciferYang** [comment]
Looks like GA is faulty, retry several times with the following error:

image

**Comment by dongjoon-hyun** [comment]
Ya, indeed. GitHub Action seems to be fail due to its own flakiness.
```
Error: Unhandled error: Error: There was a new unsynced commit pushed. Please retrigger the workflow.
[124](https://github.com/apache/spark/runs/5972998078?check_suite_focus=true#step:2:124)
    at async main (/home/runner/work/_actions/actions/github-script/f05a81df23035049204b043b50c3322045ce7eb3/dist/index.js:6147:20)
```

**Comment by LuciferYang** [comment]
https://github.com/LuciferYang/spark/actions/runs/2158563267

image

All test passed

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @LuciferYang .

**Comment by LuciferYang** [comment]
thanks @dongjoon-hyun

**Pull request closed without merging**